WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListFragment.java

```diff
@@ -462,8 +493,14 @@ public void onPostsLoaded(int postCount) {
      */
     @Override
     public void onLoadMore() {
-        if (mCanLoadMorePosts && !mIsFetchingPosts) {
-            requestPosts(true);
+        if (TextUtils.isEmpty(mSearchTerm)) {
+            if (mCanLoadMorePosts && !mIsFetchingPosts) {
+                requestPosts(true);
+            }
+        } else if (mCanSearchMore && !mIsSearching) {
+            showLoadMoreProgress();
+            mSearchOffset = mSearchResults.size();
```

Comment:
Handling the load more from here directly might be error prone, since it's not really centralized. Other devs making changes to the code might miss something here. So, what I'd suggest is to refactor the code and do something similar to `requestPosts`. That'd look something like this:
```
@Override
public void onLoadMore() {
    if (TextUtils.isEmpty(mSearchTerm)) {
        if (mCanLoadMorePosts && !mIsFetchingPosts) {
            requestPosts(true);
        }
    } else if (mCanSearchMore && !mIsSearching) {
        search(mSearchTerm, true);
    }
}
```